pkg/cmd/auth/auth.go

```diff
@@ -16,6 +16,8 @@ func NewCmdAuth(f *cmdutil.Factory) *cobra.Command {
 		Long:  `Manage gh's authentication state.`,
 	}
 
+	cmdutil.DisableAuthCheck(cmd)
```

Comment:
It looks odd that we'd have to explicitly disable the auth check for every leaf command. I realize that annotations aren't inherited, but we could still collect all annotations by traversing any command upwards:

```go
func isAuthCheckEnabled(cmd *cobra.Command) bool {
  for c := cmd; c.Parent() != nil; c = c.Parent() {
    if c.Annotations != nil && c.Annotations["skipAuthCheck"] == "true" {
      return false
    }
  }
  return true
}
```

Then we could just annotate commands immediately under root

Reply:
An observation is that any command that uses `EnableRepoOverride` is one that will always want authentication; we could disable the auth check via a `PersistentPreRun`, which children would inherit.

What do you think about that?

Reply:
actually nevermind; that would make it hard to disable the auth check for root :(